extension/src/popup/views/ReviewAuth/index.tsx

```diff
@@ -216,69 +273,127 @@ const AuthDetail = ({
     (detail) => detail.type === "sac",
   ) as FnArgsCreateSac[];
 
+  const rootJson = buildInvocationTree(rootInvocation);
+  const isInvokeContract = rootInvocation.function().switch().value === 0;
+  const transfers = isInvokeContract ? pickTransfers(rootJson) : [];
+
+  const [tokenDetails, setTokenDetails] = React.useState({} as TokenDetailMap);
+
+  const tokenDetailsUrl = React.useCallback(
+    (contractId: string) =>
+      `${INDEXER_URL}/token-details/${contractId}?pub_key=${publicKey}&network=${networkDetails.network}`,
+    [publicKey, networkDetails.network],
+  );
+
+  const transfersDepKey = JSON.stringify(transfers);
+  React.useEffect(() => {
+    async function getTokenDetails() {
+      setLoading(true);
+      const _tokenDetails = {} as TokenDetailMap;
+
+      // eslint-disable-next-line
```

Comment:
@piyalbasu this is one the rules that I would vote that we drop. 
`For..of` is the least convoluted way to do async on a list, otherwise to avoid the loop we would need to do something like an async fn in a map or for each and then `Promise.all` on them which is easier to get wrong since you it's two separate steps. 

Let me know if you disagree though.

Reply:
Yeah, I agree - I don't love the `Promise.all` paradigm either. I think we have a good oppty now to update some of our linting rules. I'd think we can drop this rule as well as the similar `no await inside of a for loop`